tuf/api/metadata.py

```diff
@@ -935,6 +963,34 @@ def get_key(self, keyid: str) -> Key:  # noqa: D102
 
         return self.keys[keyid]
 
+    def get_root_verification_result(
+        self,
+        other: "Root",
```

Comment:
It would be convenient to allow passing `None` for `other` and default to `self`. 

Plus we could check if the versions are correct:
```python
if self.version == 1 and other.version != 1 or self.version > 1 and other.version != self.version - 1:
   raise ValueError 
```

Reply:
Similarly, if we decide to make the interface version-aware, I suggest to rename `other` to `previous`.

Reply:
that sounds reasonable

Reply:
oops, I didn't do any version checking yet. I will add some of it...